helix-term/src/commands/typed.rs

```diff
@@ -2488,6 +2487,116 @@ fn read(cx: &mut compositor::Context, args: &[Cow<str>], event: PromptEvent) ->
+            ensure!(path.is_file(), "No help available for '{args_msg}'");
+            let id = editor.open(&path, Action::HorizontalSplit)?;
+            editor.document_mut(id).unwrap().set_path(None);
+
+            Ok(())
+        };
+
+    if args.is_empty() {
+        return open_help(TYPABLE_HELP_DIR, "help", cx.editor);
+    }
+
+    if args[0] == "topics" {
+        let dir_path = helix_loader::runtime_file(Path::new("help/topics"));
+
+        let callback = async move {
+            let call: job::Callback = job::Callback::EditorCompositor(Box::new(
+                move |editor: &mut Editor, compositor: &mut Compositor| {
+                    let picker = ui::file_picker(dir_path, &editor.config());
+                    compositor.push(Box::new(overlaid(picker)));
+                },
+            ));
+            Ok(call)
+        };
+        cx.jobs.callback(callback);
+
+        return Ok(());
+    }
+
+    let (help_dir, command): (&str, &str) = {
+        let arg = &args[0];
+        if let Some(command) = arg.strip_prefix(':').and_then(|arg| {
+            TYPABLE_COMMAND_LIST.iter().find_map(|command| {
+                (command.name == arg || command.aliases.contains(&arg)).then_some(command.name)
+            })
+        }) {
+            (TYPABLE_HELP_DIR, command)
+        } else if MappableCommand::STATIC_COMMAND_LIST
+            .iter()
+            .any(|command| command.name() == arg)
+        {
+            (STATIC_HELP_DIR, arg)
+        } else {
+            let arg = arg.clone().into_owned();
+            let keys = arg
+                .parse::<KeyEvent>()
+                .map(|key| vec![key])
+                .or_else(|_| helix_view::input::parse_macro(&arg))?;
+            let callback = async move {
+                let call: job::Callback = job::Callback::EditorCompositor(Box::new(
+                    move |editor: &mut Editor, compositor: &mut Compositor| {
+                        use crate::keymap::KeymapResult;
+                        let editor_view = compositor.find::<ui::EditorView>().unwrap();
+                        let mode = editor.mode;
+                        let keymaps = &mut editor_view.keymaps;
+                        let (keys, last_key) = (&keys[..keys.len() - 1], keys.last().unwrap());
+                        keys.iter().for_each(|key| {
+                            keymaps.get(mode, *key);
+                        });
+                        let key_result = keymaps.get(mode, *last_key);
+                        let result: anyhow::Result<(&str, &str)> = match &key_result {
+                            KeymapResult::Matched(command) => match command {
+                                MappableCommand::Static { name, .. } => Ok((STATIC_HELP_DIR, name)),
+                                MappableCommand::Typable { name, .. } => {
+                                    Ok((TYPABLE_HELP_DIR, name))
+                                }
```

Comment:
```suggestion
                                }
                                MappableCommand::Macro { .. } => {
                                    Err(anyhow!("`:help` for macro keybinds is not yet supported."))
                                }
```

once this is merged with master this `match` will need a branch for macros